trade-finance-manager-ui/templates/utilisation-reports/_macros/payment-details-table-row.njk

```diff
@@ -1,13 +1,28 @@
-{% macro render(paymentDetails) %}
+{% macro render(params) %}
+  {% set paymentDetails = params.paymentDetails %}
+  {% set userCanEdit = params.userCanEdit %}
+  {% set reportId = params.reportId %}
+  {% set groupStatus = paymentDetails.feeRecordPaymentGroupStatus %}
   {% set payment = paymentDetails.payment %}
   {% set feeRecords = paymentDetails.feeRecords %}
   {% set reconciledBy = paymentDetails.reconciledBy or "-" %}
   {% set dateReconciled = paymentDetails.dateReconciled or "-" %}
+  {% set paymentAmountHtml %}
+  {% if userCanEdit and groupStatus !== 'READY_TO_KEY' and groupStatus !== 'RECONCILED' %}
+    <a href="/utilisation-reports/{{ reportId }}/edit-payment/{{ payment.id }}">
+      {{ payment.amount.formattedCurrencyAndAmount }}
+    </a>
+  {% else %}
```

Comment:
I think I would prefer it if the logic of which to display as a link lived in the javascript, and then the view model had a field on paymentDetails that is the `editPaymentLink` which is either a string or undefined and we make it a link if it is defined and the userCanEdit here? That way we can use the status constant for the logic rather than hardcoding it here. What do you think, is there a reason for preferring this?

Reply:
Both approaches are valid and I don't have a preference one way or the other. The reason for having the status check at this point was purely to match the way our existing templates perform this logic (we do a similar thing on the premium payments table for rendering links), so I think we should move both sets of logic to the view model if we prefer the `editPaymentLink?: string | undefined` approach to keep this consistent